### PR TITLE
Feat/test copy next

### PR DESCRIPTION
**Pull request by JulienChampagnol**

### CODE REVIEW

File: .github/workflows/CICD.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        with:
+          node-version: 20
+
+      - name: Install dependencies
+        if: steps.semantic.outputs.new_release_published == 'true'
+        run: |
+          npm install
+          npm run package
+          npm run make
+          npm run publish --prerelease
```
Comment (BotellaA) [inline .github/workflows/CICD.yaml:39]: Why is it a prerelease ?

File: electron-server/install_back.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+rm -rf venv_back dist_back build
+python3 -m venv venv_back
+./venv_back/bin/pip install pip-tools
+./venv_back/bin/python -m pip-compile ./electron-server/requirements_back.in
+./venv_back/bin/pip install -r ./electron-server/requirements_back.txt
+./venv_back/bin/pip install pyinstaller
+source ./venv_back/bin/activate
+./venv_back/bin/pyinstaller --onefile --collect-all opengeodeweb_back ./venv_back/bin/geodeapp_back --distpath ./dist_back
```
Comment (BotellaA) [inline electron-server/install_back.sh:8]: why collect-all and not collect-data ?

File: electron-server/install_viewer.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+rm -rf venv_viewer dist_viewer build
+python3 -m venv venv_viewer
+./venv_viewer/bin/pip install pip-tools
+./venv_viewer/bin/python -m pip-compile ./electron-server/requirements_viewer.in
+./venv_viewer/bin/pip install -r ./electron-server/requirements_viewer.txt
+./venv_viewer/bin/pip install pyinstaller
+source ./venv_viewer/bin/activate
+./venv_viewer/bin/pyinstaller --onefile --collect-all opengeodeweb_viewer --collect-all vtkmodules ./venv_viewer/bin/geodeapp_viewer --distpath ./dist_viewer
```
Comment (BotellaA) [inline electron-server/install_viewer.sh:8]: collect-data?

File: electron/main.js [hunk 4]
```diff
@@ -1,24 +1,122 @@
...
+    });
+  });
+}
+
 app.whenReady().then(() => {
   const win = new BrowserWindow({
     title: "Vease - New project",
     icon: "public/favicon.ico",
     center: true,
+    width: 800,
```
Comment (BotellaA) [inline electron/main.js:81]: Could it be fullscreen ?

File: electron/main.js [hunk 5]
```diff
@@ -1,24 +1,122 @@
...
   });
+
+  win.setMinimumSize(800, 600);
+
+  ipcMain.handle("run_back", async (event, ...args) => {
+    const port = await getAvailablePort(args[0]);
+    console.log("BACK PORT", port);
+    await run_script(
+      win,
+      "./dist_back/geodeapp_back",
```
Comment (BotellaA) [inline electron/main.js:99]: maybe pyinstaller dist folder should be the same for both microservices ? Will it simplify the launch and package if both exe are in the same folder ?

File: electron/preload.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+const { contextBridge, ipcRenderer } = require("electron");
+
+console.log("_____TEST ELECTRON PRELOAD_______");
```
Comment (BotellaA) [inline electron/preload.js:3]: to remove ?

File: forge.config.cjs [hunk 7]
```diff
@@ -1,7 +1,12 @@
 module.exports = {
   packagerConfig: {
-    asar: true,
-    extraResource: ["./.output/public/", "./dist-electron"],
+    asar: false,
+    extraResource: [
```
Comment (BotellaA) [inline forge.config.cjs:4]: Should we had the pyinstaller dist folder(s)?

File: nuxt.config.js [hunk 8]
```diff
@@ -27,8 +18,22 @@ export default defineNuxtConfig({
         // Main-Process entry file of the Electron App.
         entry: "electron/main.js",
       },
+      {
+        entry: "electron/preload.js",
+        onstart(args) {
+          args.reload();
+        },
```
Comment (BotellaA) [inline nuxt.config.js:25]: What is this?

File: nuxt.config.js [hunk 9]
```diff
@@ -27,8 +18,22 @@ export default defineNuxtConfig({
...
+      },
     ],
-    renderer: {},
+  },
+
+  build: {
+    extend(config, { isClient }) {
+      // Extend only webpack config for client-bundle
+      if (isClient) {
+        config.target = "electron-renderer";
```
Comment (BotellaA) [inline nuxt.config.js:34]: Do you have documentation for this?

**Review by BotellaA — APPROVED** [review]

File: nuxt.config.js (hunk 8 above)
Comment (JulienChampagnol) [reply nuxt.config.js:25]: https://github.com/caoxiemeihao/nuxt-electron/blob/d7cf10ddaf8e050564d46b0a906cc7a468573e60/quick-start/nuxt.config.ts#L13

**Pull request merged**